numpy/lib/function_base.py

```diff
@@ -1791,7 +1791,7 @@ def place(arr, mask, vals):
     vals : 1-D sequence
         Values to put into `a`. Only the first N elements are used, where
         N is the number of True values in `mask`. If `vals` is smaller
-        than N it will be repeated.
+        than N it will be repeated. This sequence must be non-empty.
```

Comment:
May be mention that it "must be non-empty unless mask is all False", with some more thought given to the wording?